sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuSortExec.scala

```diff
@@ -284,7 +283,7 @@ case class GpuOutOfCoreSortIterator(
     // Protect ourselves from large rows when there are small targetSizes
     val targetRowCount = Math.max((targetBatchSize/averageRowSize).toInt, 1024)
 
-    if (sortedOffset == rows - 1) {
+    if (sortedOffset == rows) {
```

Comment:
I think the code would read easier if we renamed sortedOffset to sortedRows or numSortedRows

Reply:
I personally think of it in terms of offsets instead of number of rows. The fact that they end up being equal is just because the sorted values are at the first part of the batch.

Reply:
Not a blocking issue. 

Just to explain my thingking: I think of an offset as the start position of an array, or a range the way it's used on [L300]( https://github.com/NVIDIA/spark-rapids/pull/2209/files#diff-da27912783f99ffd588b32b729fec7de13fb138238df012e5e2afb87e4b483daR300)  `[sortedOffset, rows)`. Since it describes the unsorted range,  if we wanted to use the term offset, I'd call it unsortedOffset.  On the other hand, we can view it as the definition of the sorted area `[0, sortedOffset)` in which case `sortedRows` works better for me.